lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -2134,6 +2134,78 @@ static void optimizeQuantization(Function *F) {
   optimizeQuantizedMaxSplat(F);
 }
 
+#define BOILER_PLATE_TEMPLATE(NODE_NAME_)                                      \
+  template <typename Q = T, typename... ArgsTypes>                             \
+  static boilerPlateResolve<NODE_NAME_##Node, Q> *
+
+template <typename T> struct NodeCreator {
+  template <typename CmpType, typename InstType>
+  using boilerPlateResolve =
+      typename std::enable_if<std::is_same<InstType, CmpType>::value,
+                              InstType>::type;
+  BOILER_PLATE_TEMPLATE(AvgPool)
+  create(Function &F, ArgsTypes... var2) { return F.createAvgPool(var2...); }
+  BOILER_PLATE_TEMPLATE(MaxPool)
+  create(Function &F, ArgsTypes... var2) { return F.createMaxPool(var2...); }
+  BOILER_PLATE_TEMPLATE(Add)
+  create(Function &F, ArgsTypes... var2) { return F.createAdd(var2...); }
+  BOILER_PLATE_TEMPLATE(Sub)
+  create(Function &F, ArgsTypes... var2) { return F.createSub(var2...); }
+  BOILER_PLATE_TEMPLATE(Mul)
+  create(Function &F, ArgsTypes... var2) { return F.createMul(var2...); }
+  BOILER_PLATE_TEMPLATE(Div)
+  create(Function &F, ArgsTypes... var2) { return F.createDiv(var2...); }
+  BOILER_PLATE_TEMPLATE(Min)
+  create(Function &F, ArgsTypes... var2) { return F.createMin(var2...); }
+  BOILER_PLATE_TEMPLATE(Max)
+  create(Function &F, ArgsTypes... var2) { return F.createMax(var2...); }
```

Comment:
(I'm not sure this is a good idea, but...) If you wanted to really code golf this you could introduce an X-macro
```
#define CREATE(NAME) \
  template<typename T, typename ...Args> \
  enable_if_same_t<T, NAME ## Node> *create(Function &F, Args... args) { \
    return F.create ## NAME(args...); \
  }

CREATE(AvgPool)
...
#undef CREATE
```

(Yeah, ok maybe that's too baroque.)